arm simd compilation is broken

```
    /// Converts these packed floats to integers via rounding.
    #[inline]
    pub fn to_i32x4(self) -> I32x4 {
        unsafe { I32x4(round_v4f32(simd_cast(self.0))) }
    }
```
This yields:
```
error[E0308]: mismatched types
   --> C:\Users\image\pathfinder\simd\src\arm\mod.rs:335:24
    |
335 |         unsafe { I32x4(round_v4f32(simd_cast(self.0))) }
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `std::arch::aarch64::int32x4_t`, found struct `std::arch::aarch64::float32x4_t`
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -9,6 +9,7 @@ addons:
 script:
   - cargo build
   - cargo test
+  - cargo build --target  aarch64-unknown-linux-gnu
 env:
   global:
     - HARFBUZZ_SYS_NO_PKG_CONFIG=true

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,4 +1,7 @@
 language: rust
+rust:
+  - nightly
+  - stable
 addons:
   apt:
     packages:
@@ -7,13 +10,14 @@ addons:
     - libsdl2-dev
     - cmake
 script:
-  - cargo build
-  - cargo test
-  - rustup target add aarch64-unknown-linux-gnu
   - cd simd
   - cargo build --target aarch64-unknown-linux-gnu
   - cd ../geometry
   - cargo build --target  aarch64-unknown-linux-gnu
+  - cd ..
+  - cargo build
+  - cargo test
+  - rustup target add aarch64-unknown-linux-gnu
 env:
   global:
     - HARFBUZZ_SYS_NO_PKG_CONFIG=true

--- a/.travis.yml
+++ b/.travis.yml
@@ -10,6 +10,9 @@ script:
   - cargo build
   - cargo test
   - rustup target add aarch64-unknown-linux-gnu
+  - cd simd
+  - cargo build --target aarch64-unknown-linux-gnu
+  - cd ../geometry
   - cargo build --target  aarch64-unknown-linux-gnu
 env:
   global:

--- a/.travis.yml
+++ b/.travis.yml
@@ -10,6 +10,7 @@ addons:
     - libsdl2-dev
     - cmake
 script:
+  - rustup target add aarch64-unknown-linux-gnu
   - cd simd
   - cargo build --target aarch64-unknown-linux-gnu
   - cd ../geometry
@@ -17,7 +18,6 @@ script:
   - cd ..
   - cargo build
   - cargo test
-  - rustup target add aarch64-unknown-linux-gnu
 env:
   global:
     - HARFBUZZ_SYS_NO_PKG_CONFIG=true

--- a/.travis.yml
+++ b/.travis.yml
@@ -9,6 +9,7 @@ addons:
 script:
   - cargo build
   - cargo test
+  - rustup target add aarch64-unknown-linux-gnu
   - cargo build --target  aarch64-unknown-linux-gnu
 env:
   global:

--- a/geometry/src/vector.rs
+++ b/geometry/src/vector.rs
@@ -246,6 +246,14 @@ impl Div<Vector2F> for Vector2F {
     }
 }
 
+impl Div<f32> for Vector2F {
+    type Output = Vector2F;
+    #[inline]
+    fn div(self, other: f32) -> Vector2F {
+        self / Vector2F::splat(other)
+    }
+}
+
 impl Neg for Vector2F {
     type Output = Vector2F;
     #[inline]

--- a/simd/src/arm/mod.rs
+++ b/simd/src/arm/mod.rs
@@ -13,7 +13,7 @@ use std::arch::aarch64::{uint32x2_t, uint32x4_t};
 use std::f32;
 use std::fmt::{self, Debug, Formatter};
 use std::mem;
-use std::ops::{Add, BitAnd, BitOr, Index, IndexMut, Mul, Shr, Sub};
+use std::ops::{Add, BitAnd, BitOr, Div, Index, IndexMut, Mul, Shr, Sub};
 
 mod swizzle_f32x4;
 mod swizzle_i32x4;

--- a/simd/src/arm/mod.rs
+++ b/simd/src/arm/mod.rs
@@ -559,12 +559,12 @@ impl I32x4 {
 
     #[inline]
     pub fn max(self, other: I32x4) -> I32x4 {
-        unsafe { I32x4(simd_fmax(self.0, other.0)) }
+        unsafe { I32x4(simd_cast(simd_fmax(self.to_f32x4().0, other.to_f32x4().0))) }
     }
 
     #[inline]
     pub fn min(self, other: I32x4) -> I32x4 {
-        unsafe { I32x4(simd_fmin(self.0, other.0)) }
+        unsafe { I32x4(simd_cast(simd_fmin(self.to_f32x4().0, other.to_f32x4().0))) }
     }
 
     // Packed comparisons

--- a/simd/src/arm/mod.rs
+++ b/simd/src/arm/mod.rs
@@ -332,7 +332,7 @@ impl F32x4 {
     /// Converts these packed floats to integers via rounding.
     #[inline]
     pub fn to_i32x4(self) -> I32x4 {
-        unsafe { I32x4(round_v4f32(simd_cast(self.0))) }
+        unsafe { I32x4(simd_cast(round_v4f32(self.0))) }
     }
 }
 

--- a/simd/src/arm/mod.rs
+++ b/simd/src/arm/mod.rs
@@ -445,6 +445,18 @@ impl I32x2 {
         unsafe { U32x2(simd_eq(self.0, other.0)) }
     }
 
+    // Basic operations
+
+    #[inline]
+    pub fn max(self, other: I32x2) -> I32x2 {
+        self.to_i32x4().max(other.to_i32x4()).xy()
+    }
+
+    #[inline]
+    pub fn min(self, other: I32x2) -> I32x2 {
+        self.to_i32x4().min(other.to_i32x4()).xy()
+    }
+
     // Concatenations
 
     #[inline]
@@ -458,6 +470,11 @@ impl I32x2 {
     #[inline]
     pub fn to_f32x2(self) -> F32x2 {
         unsafe { F32x2(simd_cast(self.0)) }
+    }
+    
+    #[inline]
+    pub fn to_i32x4(self) -> I32x4 {
+        self.concat_xy_xy(I32x2::default())
     }
 }
 

